yt_dlp/extractor/vk.py

```diff
@@ -462,20 +463,28 @@ def _real_extract(self, url):
                 formats.append({
                     'format_id': format_id,
                     'url': format_url,
+                    'ext': 'mp4',
+                    'source_preference': 1,
                     'height': height,
                 })
             elif format_id == 'hls':
```

Comment:
`elif format_id.startswith('hls') `

Like the dash format, this should probably be startswith as well, I have come across such before though I don't have any examples.  I vaguely remember it being a live stream about a half a year ago and thus been autopatching it since.

I also have `"raise ExtractorError(clean_html(payload[0][1:-1]), expected=True)"` changed to `"raise UserNotLive(video_id=video_id)"` but I don't remember why.

Reply:
I have pushed [a commit to the cleanup PR](https://github.com/yt-dlp/yt-dlp/pull/9765/commits/97c4c860436fd4e9df52e2ee5c521844d3998ef4) for extracting additional potentially existing hls formats. There exists `hls_live_playback` on livestreams, but without replacing `offset_p` (and custom download code) those cannot be downloaded. Additionally, I excluded the live dash formats as they have a similar issue.

Also, if you do improve an extractor, feel free to open a PR instead of just patching your local instance.